tools/test/travis-ci/functions.sh

```diff
@@ -59,7 +59,7 @@ _install_gcc()
 {
   # Ignore shellcheck warnings: Variables defined in .travis.yml
   # shellcheck disable=SC2154
-  local url="${deps_url}/gcc6-linux.tar.bz2"
+  local url="${deps_url}/gcc9-linux.tar.bz2"
 
   # shellcheck disable=SC2154
   local gcc_path="${deps_dir}/gcc/gcc-arm-none-eabi-6-2017-q2-update/"
```

Comment:
you might want to change this as well

Reply:
is `arm-none-eabi-gcc` been used at all ?

Reply:
it is used for event/littlefs builds. the logs now are

```
I: URL: https://mbed-os-ci.s3-eu-west-1.amazonaws.com/jenkins-ci/deps/gcc9-linux.tar.bz2
I: Installing 'gcc'
install.2
0.01s$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (GNU Tools for ARM Embedded Processors 6-2017-q2-update) 6.3.1 20170620 (release) [ARM/embedded-6-branch revision 249437]
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
python tools/make.py -t GCC_ARM -m K64F --source=. --build=BUILD/K64F/GCC_ARM -j0
[Warning] @,: Compiler version mismatch: Have 6.3.1; expected version >= 9.0.0 and < 10.0.0
```

Reply:
@0xc0170 @jamesbeyond 
There seems to be another bug in Travis config. What is the expectation from Travis? Validate PR contents then it doesn't seems to be doing what is it intended for. I do not see it fetching the PR contents in the config. May be I'm missing something?

I'm speculating this because, Travis succeeded on this PR and the log shows https://travis-ci.org/github/ARMmbed/mbed-os/jobs/717990846 `arm-none-eabi` version 6 in the PATH which shouldn't have happened with this PR. The travis job should have failed.

Reply:
Travis didn't fail and reported `arm-none-eabi` version 6 because the version 6 was cached. 
`arm-none-eabi` version 9 is uploaded and configured in Travis and the old cache has been removed.